mne/beamformer/tests/test_lcmv.py

```diff
@@ -99,6 +99,49 @@ def _get_data(tmin=-0.1, tmax=0.15, all_forward=True, epochs=True,
         forward_surf_ori, forward_fixed, forward_vol
 
 
+@testing.requires_testing_data
+def test_lcmv_vector():
+    """Test vector LCMV solutions."""
+    info = mne.io.read_raw_fif(fname_raw).info
+    info = mne.pick_info(info, mne.pick_types(info, meg=True, exclude=()))
+    info.update(bads=[], projs=[])
+    forward = mne.read_forward_solution(fname_fwd)
+    forward = mne.pick_channels_forward(forward, info['ch_names'])
+    vertices = [s['vertno'][::100] for s in forward['src']]
+    n_vertices = sum(len(v) for v in vertices)
+    assert 5 < n_vertices < 20
+    stc = mne.SourceEstimate(100e-9 * np.eye(n_vertices), vertices,
+                             0, 1. / info['sfreq'])
+    forward_sim = mne.convert_forward_solution(forward, force_fixed=True,
+                                               use_cps=True)
+    forward_sim = mne.forward.restrict_forward_to_stc(forward_sim, stc)
+    data_cov = mne.make_ad_hoc_cov(info)
+    noise_cov = mne.make_ad_hoc_cov(info)
+    noise_cov['data'] = np.diag(noise_cov['data'])
+    evoked = mne.EvokedArray(forward_sim['sol']['data'], info)
+    with pytest.raises(ValueError, match='pick_ori must be one of'):
+        make_lcmv(evoked.info, forward, data_cov, 0.05, noise_cov,
+                  pick_ori='bad')
+    angles = list()
+    for ti in range(n_vertices):
+        data_cov['data'] = (np.outer(evoked.data[:, ti], evoked.data[:, ti]) +
+                            noise_cov['data'])
```

Comment:
Hi @larsoner ,
could the problem lie here? The noise covariance matrix and data covariance matrix have quite a different order of magnitude, which means that this does not create a diagonal matrix and thus ``data_cov`` is highly rank-deficient (rank of 1), which then creates problems when taking the inverse in ``make_lcmv``.

Reply:
I tried changing the relative orders of magnitude by changing the simulated amplitude above and it had little effect